scripts/tools/zap/run_zaptool.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+#
+# Copyright (c) 2021 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+CHIP_ROOT=$(readlink -f "${BASH_SOURCE%/scripts/tools/zap/run_zaptool.sh}")
+ZAP_ROOT="$CHIP_ROOT/third_party/zap/repo"
+[[ -n "$1" ]] && ZAP_ARGS=(-i "$(readlink -f "$1")") || ZAP_ARGS=()
+
+cd "$CHIP_ROOT" &&
+    git submodule update --init third_party/zap/repo ||
```

Comment:
how about `set -e` on the script?

That way we would not need the `|| exit 1` lines and we would also be less likely to make logic errors and forgetting in the future